### PR TITLE
Branches merged

### PR DESCRIPTION
**Pull request by malminawi**
Merged frontendDev and RegistrationEdit branches.
Added extra fields for student registration. 
Fixed the permissions, student can only see the student page, mentor can see the mentor and student pages, and faculty can see all pages.

### CODE REVIEW

File: auth_backend/templates/user/profile_mentor_info.html [hunk 1]
```diff
@@ -0,0 +1,23 @@
+{% extends 'student_base.html' %}
```
Comment (malminawi) [inline auth_backend/templates/user/profile_mentor_info.html]: Is it ok if I delete this file, as we are not using it anymore?

File: auth_backend/templates/user/profile_student_info.html [hunk 2]
```diff
@@ -0,0 +1,23 @@
+{% extends 'student_base.html' %}
```
Comment (malminawi) [inline auth_backend/templates/user/profile_student_info.html]: Is it ok if I delete this file, as we are not using it anymore?

File: auth_backend/templates/user/profile_mentor_info.html (hunk 1 above)
Comment (NullDefault) [reply auth_backend/templates/user/profile_mentor_info.html]: yup

File: auth_backend/templates/user/profile_student_info.html (hunk 2 above)
Comment (NullDefault) [reply auth_backend/templates/user/profile_student_info.html]: asbolutely

**Review by NullDefault — APPROVED** [review]

**Pull request merged**